### ISSUE TITLE
Release 2.0

### ISSUE THREAD

**Issue by digitaldan** [openhab/openhab.android#499]
@mueller-ma  , @lolodomo , @FlorianSW ,  @maniac103 

Hi all, I would like to cut 2.0 this weekend, I will plan on going through the PR's we have and merging them.   If there are PR's you think need more work, or could wait until 2.0.1, please let me know.   These are the PR's that seem to be a candidate for release that are not yet merged. 

#481 
#482 
#493 
#497 
#498 

@FlorianSW is #391 ready or does this need more testing?
@mueller-ma is #496 ready to merge or should i help test this? 

Let me know if this looks right to you, thanks!

**Comment by maniac103** [comment]
#496 and #497 look like they would need the largest testing. I also have some specific comments about #496 (will add them later), my gut feeling is we shouldn't rush out this one.

**Comment by ThomDietrich** [comment]
My believe is that @digitaldan wants to release before the openHAB 2.2 release. 👍

**Comment by maniac103** [comment]
@ThomDietrich Oh, that one is imminent? Please do me a favor and review https://github.com/openhab/openhab2-addons/pull/2932 before that happens so I can finally update to OH2 - thanks ;-)

**Comment by mueller-ma** [comment]
I preffer that my two PRs about the foss build #482 and #493 will be merged AFTER the 2.0 release. We dont need them for the full version.
#481 looked quite good to me, but I didnt review the changes after my initial review there.
#391 the last time I tried the migration it didnt work, but @FlorianSW said he fixed it.

**Comment by mueller-ma** [comment]
I tested #496, but only for local connections.

**Comment by mueller-ma** [comment]
@karafyllidis @mgarciate @lolodomo @naofum @jvitkauskas @Maurits28 
Please update the translations. I'll have a look at the german one.

**Comment by lolodomo** [comment]
#497 is very simple and don't require lots of tests. #481 should require more attention.

I am working on the fix relative to the 30 seconds useless refresh. I would like this fix to be included in the comiong 2.0. I implemented the fix very quickly this morning but unfortunately I did not notice the expected result as it avoided a refresh only every minute rather than every 30 seconds. I have to investiguate if there is something in the demo that could explain this behaviour. I will try to provide the final fix today or tomorrow (Saturday).

**Comment by mueller-ma** [comment]
I would like to see #476 resolved before a new version is built. I'd suggest to wait until it is resolved or - if we need the version this weekend - release a commit before #442 was merged.

**Comment by lolodomo** [comment]
Ok, I finally submitted PR #501 for the 30s refresh. Unfortunately, I discovered a stupid bug in my fix for the sitemap REST API. I will fix it in parallel. It would make sense to wait for the fix in server before merging this one.

**Comment by FlorianSW** [comment]
For #391 I would really like to see it in the next version (2.0) of the app. I already fixed all points and implemented all suggestions (as far as I can see), however, it would be great, if @mueller-ma could take a look to the migration part, which worked in my testing (building apk of older version -> installing -> changing url -> building apk of this version and install -> see if the url was migrated) :) Otherwise the PR should be fine.

Also: It's not a real blocker, however, it would be cool, if we could fix #479 before the next release, too, but I think we can't get it fixed before this weekend anymore :)

**Comment by lolodomo** [comment]
@digitaldan : I would suggest to publish first a version 1.9 this weekend.
I think the version 2.0 should include the suppression of the top right menu. It remains 2 entries in this menu, the RFC tag and the selection of the default sitemap. As already proposed, the selection of the default sitemap should be moved to the settings.
Then you could publish a version 2.0 just after the openHAB 2.2 is released in one week. This 2.0 would include the annoying issue causing a refresh of the app every 30 seconds.

**Comment by lolodomo** [comment]
Please include #503 an update of the French translation.

**Comment by mgarciate** [comment]
@mueller-ma do you have the German translation updated at master? I'd like to check differences between GE and SP, and fill the resource with the new ones.

**Comment by mueller-ma** [comment]
@mgarciate In the german translation only the translations for "Bitmap" and "Vector" are missing.

**Comment by mgarciate** [comment]
Alright, I'll upload the translation this evening.

**Comment by mueller-ma** [comment]
I'll added a 2.0 milestone to all issues and PRs which would be nice to have in the next version.

**Comment by boc-tothefuture** [comment]
I am going to try and get to #476 today

**Comment by boc-tothefuture** [comment]
PR #507 submitted to resolve #476

**Comment by digitaldan** [comment]
If anyone wants to start the release notes for the play store that would be great.  For example:

* Updated Logo
* New Translations
* Bug Fixes
* ....

**Comment by mueller-ma** [comment]
On it

**Comment by digitaldan** [comment]
Thanks!

**Comment by mueller-ma** [comment]
I think we should also add it in this repo #510

**Comment by mgarciate** [comment]
Spanish translation in #512

**Comment by jvitkauskas** [comment]
Lithuanian, Russian translations in https://github.com/openhab/openhab.android/pull/513

**Comment by lolodomo** [comment]
Kai is building a new server version including the REST API fix so please include #501 in version 2.0.

**Comment by karafyllidis** [comment]
Greek translations in #518

**Comment by digitaldan** [comment]
FYI I just pushed a change so every merge into master will trigger the build AND push a signed APK file to https://github.com/openhab/openhab.android/releases/tag/latest-master  if the build is successful

**Comment by digitaldan** [comment]
well, it was working!  trying to debug why its now not updating, stay tuned.

In the mean time #499 and #510 are the only un-merged PR's.  I'll merge #499 here in just a min, @mueller-ma that should complete our final code change for the build, if we are happy we can merge the readme and tag the release.

**Comment by lolodomo** [comment]
Please include  #501.

**Comment by digitaldan** [comment]
Just merged #501

**Comment by lolodomo** [comment]
Thank you

**Comment by mueller-ma** [comment]
#516 is the last PR for 2.0

**Comment by digitaldan** [comment]
Final APK is here  https://github.com/openhab/openhab.android/releases/tag/2.0.0-PRE , I start setting up the play store for release, I'll post back here however before I push anything.

**Comment by digitaldan** [comment]
Well, we have a 500 character limit for notes in the play store for each language, so we are going to need a smaller list, this is the format. 

```
<en-US>
<h3>Added</h3>

• Add adaptive app icon
• Add two new themes
• Add app intro
• Add about screen
• Implement support for HSL videos
• Add theme support for charts. Need openHAB server > 2.1
• Added separate user and password settings for local and remote connection

<h3>Changed</h3>

• Update app icon
• Improved selection and setpoint items
• Move settings to the side menu
• Improved error messages
• Improved charts size. Need openHAB server > 2.1
• Better handling of image items
• Right align selection value
• General performance improvements
• Update translations

<h3>Fixed</h3>

• Fixed overlapping label and value
• Fixed a bug where images and charts were not shown
• Fixed a bug where notification were not shown when on WIFI
• Fixed a bug where no charts was shown when icon type set to SVG
• Fixed some bugs where wrong icon was shown
• Removed useless sitemap refreshes

<h3>Removed</h3>

• Remove support for controlling items via volume buttons
• Remove Bindings and Discovery funtionality. Please use PaperUI instead
</en-US>

<de-DE>
Enter or paste your release notes for de-DE here
</de-DE>

```

I can start working on this in a few hours

**Comment by mueller-ma** [comment]
```
<en-US>
<h3>Added</h3>

• Add new themes
• Add app intro
• Added separate user and password settings for local and remote connection

<h3>Changed</h3>

• Improved selection and setpoint items
• General performance improvements
• Update translations

<h3>Fixed</h3>

• Fixed overlapping label and value
• Fixed a bug where images and charts were not shown
• Fixed some bugs where wrong icon was shown
• Removed useless sitemap refreshes
</en-US>

<de-DE>
<h3>Neu</h3>

* App-Intro
* Neue Themes
* Unterschiedliche Nutzer für lokalen und remote Server werden jetzt unterstützt

<h3>Änderungen</h3>

* Bessere Darstellung von Selection und Setpoint

<h3>Behoben</h3>
* Überlappende Label und Werte wurden behoben
* Ein Fehler wurde behoben, bei dem Diagramme und Bilder nicht angezeigt wurden
* Fehler wurden behoben, bei denen falsche Icons angezeigt wurden
* Unötiges Nachladen der Sitemap wurde entfernt
</de-DE>

```

You mostly need [more words](https://esolangs.org/wiki/German#Limitations) in german than in english, so I removed some points from the german changelog, e.g. translation (They dont care in germany anyway, since the german translation was nearly complete). @maniac103 Please review. Whats short and good for `Fixed overlapping label and value`.

**Comment by maniac103** [comment]
```
* Neue Themes
* Unterschiedliche Nutzer für lokalen und remote Server werden jetzt unterstützt
```

I'd write as
```
* Neue Designs
* Unterstützung von unterschiedlichen Anmeldedaten für lokalen und entfernten Server
```
`Remote-Server` instead of `entfernten Server` would also be ok.

For `Fixed overlapping label and value` I'd suggest `Überlagerung von Bezeichnungen und Werten bei langen Bezeichnungen behoben`. I'd also write `Unnötiges Nachladen der Sitemap unter bestimmten Bedingungen wurde behoben`

**Comment by mueller-ma** [comment]
@maniac103 Thanks
@digitaldan Changelog above is ready for play store.

**Comment by digitaldan** [comment]
Thanks @mueller-ma ! looks great!

**Comment by digitaldan** [comment]
```
Release notes for de-DE should be within the 500 character limit.
```
Looks like we are still over on the German part.

**Comment by mueller-ma** [comment]
Remove that part with Unnötiges nachladen...

**Comment by digitaldan** [comment]
All right, I have the latest apk loaded in the play store , release notes are entered,  speak now or forever hold your peace!

**Comment by mueller-ma** [comment]
... or at least until 2.0.1 :)

**Comment by FlorianSW** [comment]
Hmm, should we probably wait for #527? :/

**Comment by digitaldan** [comment]
https://github.com/openhab/openhab.android/releases/tag/2.0.0-PRE-2 is now available with #527, I will need to make one more PR to bump the build number, but this can be tested to verify the fix

**Comment by lolodomo** [comment]
The menu at top right has now disappeared. That is really cool like that.

**Comment by digitaldan** [comment]
I have posted https://github.com/openhab/openhab.android/releases/tag/2.0.0 , I will plan on releasing this in a few hours unless I hear otherwise.  I will also post an announcement on the community forums.  Thanks for everyones help!

**Comment by digitaldan** [comment]
https://community.openhab.org/t/android-and-ios-2-0-releases 

Thanks everyone!

**Comment by lolodomo** [comment]
With the last minute changes, the translations are now only partial in the preference settings.

**Comment by FlorianSW** [comment]
@digitaldan Could you remove the bindings and discovery screenshot from the list of screenshots? These functionalities were removed :)

**Comment by digitaldan** [comment]
Good point, I’ll do that now!

**Comment by mueller-ma** [comment]
There are still some binding screenshots in the play store.

**Comment by mueller-ma** [comment]
The reviews on the german and english play store are all 4 to 5 stars for this release, except one user giving us one star for removing the slider volume button control.

**Comment by boc-tothefuture** [comment]
> except one user giving us one star for removing the slider volume button control.

https://xkcd.com/1172/

:)

**Comment by mueller-ma** [comment]
There is always a xkcd :)

**Comment by kaikreuzer** [comment]
> one user giving us one star for removing the slider volume button control

And it wasn't me, although i am missing that feature as well ;-) But no worries, I won't give a negative rating just because of such a small feature 😎
